test/csx.coffee

```diff
@@ -0,0 +1,616 @@
+# We usually do not check the actual JS output from the compiler, but since
+# CSX is not readily supported by Node, we do it in this case
+eqCSX = (cs, js) ->
+  eq CoffeeScript.compile(cs, {bare: true}), js + '\n'
+
+test 'self closing', ->
+  eqCSX '''
+    <div />
+  ''', '''
+    <div />;
+  '''
+
+test 'self closing formatting', ->
+  eqCSX '''
+    <div/>
+  ''', '''
+    <div />;
+  '''
+
+test 'self closing multiline', ->
+  eqCSX '''
+    <div
+    />
+  ''', '''
+    <div />;
+  '''
+
+test 'regex attribute', ->
+  eqCSX '''
+    <div x={/>asds/} />
+  ''', '''
+    <div x={/>asds/} />;
+  '''
+
+test 'string attribute', ->
+  eqCSX '''
+    <div x="a" />
+  ''', '''
+    <div x="a" />;
+  '''
+
+test 'simple attribute', ->
+  eqCSX '''
+    <div x={42} />
+  ''', '''
+    <div x={42} />;
+  '''
+
+test 'assignment attribute', ->
+  eqCSX '''
+    <div x={y = 42} />
+  ''', '''
+    var y;
+
+    <div x={y = 42} />;
+  '''
+
+test 'object attribute', ->
+  eqCSX '''
+    <div x={{y: 42}} />
+  ''', '''
+    <div x={{
+        y: 42
+      }} />;
+  '''
+
+test 'paired', ->
+  eqCSX '''
+    <div></div>
+  ''', '''
+    <div></div>;
+  '''
+
+test 'simple content', ->
+  eqCSX '''
+    <div>Hello world</div>
+  ''', '''
+    <div>Hello world</div>;
+  '''
+
+test 'content interpolation', ->
+  eqCSX '''
+    <div>Hello {42}</div>
+  ''', '''
+    <div>Hello {42}</div>;
+  '''
+
+test 'nested tag', ->
+  eqCSX '''
+    <div><span /></div>
+  ''', '''
+    <div><span /></div>;
+  '''
+
+test 'tag inside interpolation formatting', ->
+  eqCSX '''
+    <div>Hello {<span />}</div>
+  ''', '''
+    <div>Hello <span /></div>;
+  '''
+
+test 'tag inside interpolation, tags are callable', ->
+  eqCSX '''
+    <div>Hello {<span /> x}</div>
+  ''', '''
+    <div>Hello {<span />(x)}</div>;
+  '''
+
+test 'tags inside interpolation, tags trigger implicit calls', ->
+  eqCSX '''
+    <div>Hello {f <span />}</div>
+  ''', '''
+    <div>Hello {f(<span />)}</div>;
+  '''
+
+test 'regex in interpolation', ->
+  eqCSX '''
+    <div x={/>asds/}><div />{/>asdsad</}</div>
+  ''', '''
+    <div x={/>asds/}><div />{/>asdsad</}</div>;
+  '''
+
+# Unlike in coffee-react-transform
+test 'bare numbers not allowed', ->
+  throws -> CoffeeScript.compile '<div x=3 />'
```

Comment:
There's no reason to have these test both here and in error_messages.coffee, is it?